Added support for a JSON health endpoint

aleksihakli opened the pull request:
Rebased #31 on top of the current master to get it finally merged.

Commit:
Added support for a JSON health endpoint

aleksihakli committed:
Add tests for health check views

codingjoe commented on health_check/urls.py:
```diff
@@ -2,10 +2,11 @@
 
 urlpatterns = [
     url(r'^$', home, name='health_check_home'),
+    url(r'^json/?$', json_view, name='health_check_json_view')
```
What do you think of not add a separate path but making the view header sensitive? You could request the response to be a json.
```
Accept: application/json
```

aleksihakli replied:
That's a good idea. I'll just add a different view per Accept header and add some documentation :)

codingjoe commented on health_check/backends/base.py:
```diff
@@ -15,6 +15,12 @@ class HealthCheckStatusType(object):
     2: _("unexpected result"),
 }
 
+HEALTH_CHECK_JSON_STATUS_TYPE_TRANSLATOR = {
```
This could be an enum.

aleksihakli replied:
Should we refactor the original `HEALTH_CHECK_STATUS_TYPE_TRANSLATOR` as well, then? The current implementation uses the same logic for both; look up pretty response status based on the status code returned by the backend.

codingjoe commented on tests/test_views.py:
```diff
@@ -0,0 +1,44 @@
+
+try:
+    # Django >= 1.4, < 2.0 import
+    from django.core.urlresolvers import reverse
```
No need to add support for any version prior to 1.8

codingjoe reviewed (changes requested):
In general I like the idea.
I don't know if it's not better to use literals instead of integers in the JSON.

Besides that, you really should add some documentation here.

aleksihakli committed:
Refactor health_check_home to respond with different types based on HTTP Accept header

aleksihakli commented:
@codingjoe I refactored the view to return the status with the `Accept: application/json` header and added tests for the JSON output. I'll rebase the commit chain on top of the 2.0 rewrite later to keep the old attributions here.

aleksihakli commented:
Up-to-date PR in #80.
